entrypoint/upgradecluster/upgradenodereplacement_test.go

```diff
@@ -80,8 +82,35 @@ var _ = Describe("Test:", func() {
 			testcase.TestServiceLoadBalancer(false, true)
 		})
 	}
+
+	AfterAll(func() {
+		deleteAWSResources()
+	})
+		
 })
 
+func deleteAWSResources() {
```

Comment:
can we add that on suite file? just to keep those files as clean as possible, without any logics ?

Reply:
also on suite if we can create aws client on main and pass it through like we do with cluster,flags etc... just to keep the pattern of initializing dependencies on main.

Reply:
so I tried doing that, not sure why but it seems to fetch the node external ip from the previous cluster

Reply:
Also cannot call it on the suite as the suite applies to all the test and this func is needed only for this test

>  agreed

Reply:
thats really odd, cause the `aws.AddClient` has nothing to do with cluster.IPS stuff.

Reply:
but thats ok we can look at that later

Reply:
Yeah I was confused as well as to why it was not working like that

Reply:
sorry ....really not trying to be picky , but this is like we are doing something that we fought not do to until today lol.....adding logic on  IT specs entrypoint.....

can we figure it out a way do call use this on the logic place? since you are adding this as a specific helper for node replace, can we add this on `/pkg/testcase/nodereplacement.go` ? 

i know that adding there will be a lit bit of trick , but sure is doable

Reply:
@mdrahman-suse

Reply:
Let me try

Reply:
Updated, please take a look @fmoral2

Reply:
sorry for all . let me try again 


i wanted to mean that it should not live on  `/entrypoint/...`  on a IT spec , since its not a test or nor a feature, and even adding there a just a after something was polluting that file , which in theory should be like a file for " outsiders" to see whats been tested on a feature/flow without needing to code on actual code 
not saying that i like it, just saying its purpose.


since we are treating that ( and i agree ) as a upgrade node replacement helper , as other ones like 
`delErr := deleteRemainServer(serverLeaderIP, awsDependencies)`
`func nodeReplaceServers(`
`serverJoin()`
and etc, this should live on the same file or as closest as possible.


- Adding that as a aws utils does not suite better , because the idea of a client pkg, is kind of  "provide an interface and methods of something, so you can use it wherever its needed" so you probably should be exporting this `DeleteEC2Instances()` as a method, not as a util func, here util func would be util or helpers for the pkg itself, not for  outside , similar to this here 
- https://github.com/rancher/distros-test-framework/pull/177/files#diff-d050ce959548e8086404b5d8d53e4f2ddce5917e83a082106bd34416b306244b

And adding that as a method would not make much sense cause we already have a method exported `DELETEINSTANCE`


Thats why where it would be fitting nicer is on the actual test case , because there you are importing a client to make use of it ( aws) and based on methods provided ( DELETE INSTANCE) you can use it and create a custom helper for your needs `DeleteEC2Instances()`

Reply:
So if we add the func in the same file, when and where should we call the func?